pkg/collector/corechecks/sbom/convert.go

```diff
@@ -293,9 +293,17 @@ func convertDependency(in *cyclonedx.Dependency) *cyclonedx_v1_4.Dependency {
 		return nil
 	}
 
+	var dependencies []*cyclonedx_v1_4.Dependency
+
+	for _, dep := range *in.Dependencies {
+		dependencies = append(dependencies, &cyclonedx_v1_4.Dependency{
+			Ref: dep,
+		})
+	}
+
 	return &cyclonedx_v1_4.Dependency{
 		Ref:          in.Ref,
-		Dependencies: convertArray(in.Dependencies, convertDependency),
+		Dependencies: dependencies,
 	}
```

Comment:
This would be more elegant: https://github.com/DataDog/datadog-agent/pull/15081/files#diff-7e8ac7b0c4a259b6fff66f3bd7b9b68098495cb4c1a3804e89dda8bee9bfcfa0R297-R311

Reply:
Fixed. I've rebased on top of main to pick your change :+1: